docs/posts/flink-quickstart/index.qmd

```diff
@@ -0,0 +1,262 @@
+
+```{python}
+import ibis
+from pyflink.table import EnvironmentSettings, TableEnvironment
+
+env_settings = EnvironmentSettings.in_streaming_mode()
+table_env = TableEnvironment.create(env_settings)
+# write all the data to one file
+table_env.get_config().set("parallelism.default", "1")
+
+connection = ibis.flink.connect(table_env)
+```
+
+Flink’s streaming connectors aren't part of the binary distribution. Link the [Kafka connector](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/connectors/table/kafka/) for cluster execution by adding the JAR file from the cloned repository.
+
+```{python}
+#| echo: false
+#| output: false
+
+import urllib.request
+
+urllib.request.urlretrieve(
+    "https://repo.maven.apache.org/maven2/org/apache/flink/flink-sql-connector-kafka/3.0.2-1.18/flink-sql-connector-kafka-3.0.2-1.18.jar",
+    filename="flink-sql-connector-kafka-3.0.2-1.18.jar",
+)
+```
+
+```{python}
+#| output: false
+
+connection._exec_sql("ADD JAR 'flink-sql-connector-kafka-3.0.2-1.18.jar'")
+```
+
+#### Create the source and sink tables
+
+Use [`create_table`](../../backends/flink.qmd#ibis.backends.flink.Backend.create_table) to register tables. Notice the new top-level `ibis.watermark` API for [specifying a watermark strategy](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/concepts/time/#event-time-and-watermarks).
+
+```{python}
+import ibis.expr.datatypes as dt
+import ibis.expr.schema as sch
+
+# create source Table
+source_schema = sch.Schema(
+    {
+        "createTime": dt.timestamp(scale=3),
+        "orderId": dt.int64,
+        "payAmount": dt.float64,
+        "payPlatform": dt.int32,
+        "provinceId": dt.int32,
+    }
+)
+
+source_configs = {
+    "connector": "kafka",
+    "topic": "payment_msg",
+    "properties.bootstrap.servers": "localhost:9092",
+    "properties.group.id": "test_3",
+    "scan.startup.mode": "earliest-offset",
+    "format": "json",
+}
+
+t = connection.create_table(
+    "payment_msg",
+    schema=source_schema,
+    tbl_properties=source_configs,
+    watermark=ibis.watermark(
+        time_col="createTime", allowed_delay=ibis.interval(seconds=15)
+    ),
+)
+
+# create sink Table
+sink_schema = sch.Schema(
+    {
+        "province_id": dt.int32,
+        "pay_amount": dt.float64,
+    }
+)
+
+sink_configs = {
+    "connector": "kafka",
+    "topic": "sink",
+    "properties.bootstrap.servers": "localhost:9092",
+    "format": "json",
+}
+
+connection.create_table(
+    "total_amount_by_province_id", schema=sink_schema, tbl_properties=sink_configs
+)
+```
+
+#### Perform calculations
+
+Compute the total pay amount per province in the past 10 seconds (as of each message, for the province in the incoming message).
+
+```{python}
+agged = t[
+    t.provinceId.name("province_id"),
+    t.payAmount.sum()
+    .over(
+        range=(-ibis.interval(seconds=10), 0),
+        group_by=t.provinceId,
+        order_by=t.createTime,
+    )
+    .name("pay_amount"),
+]
```

Comment:
```suggestion
agged = t.select(
    province_id=t.provinceId,
    pay_amount=t.payAmount.sum().over(
        range=(-ibis.interval(seconds=10), 0),
        group_by=t.provinceId,
        order_by=t.createTime,
    )
)
```

Since these names are static the `select` method is much less verbose here.